### ISSUE TITLE
Implement demo and document how to convey progress bar color meanings

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3517]
## Describe the bug

Progress bars convey success and danger status by color alone, which is not accessible for all users.

## How to reproduce

https://clarity.design/documentation/progress

- Using a screen reader read the Success and Danger Progress Bars and observe that they do not indicate what they represent.

## Expected behavior

Add a visible textual label that uses aria-labelledby to programmatically associate the label with the progress bar as seen below

<label id="l1">Percent complete</label>
<progress id="p1" aria-labelledby="l1" value="30" max="100"></progress>

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

This is a demo and website feature, as apps will have to apply this logic themselves. If the label is added, it might be necessary to review the design as this changes the display and perhaps the recommendations we put on the website. Ensure this is cleared with design before implementation.

**Comment by mathisscott** [comment]
@chlane 
If the recommendation is to allow for a label to be used so that the color alone does not convey the info that the progress bar represents, is there any objection to making the appearance of the label a screen-reader only thing?

**Comment by chlane** [comment]
@mathisscott 
No. If I color blind user, not using a screen reader to assist them in reading the page, saw these progress bars, they would not be able to tell which one represents Success vs. which one represents Danger.

**Comment by mathisscott** [comment]
Got it. Thanks, @chlane !

Mkay. So the recommendation here should be: 

* a label always be included with a progress bar
* We will need to use `aria-labelledby` along with our id generator to tie the label and the progress bar together. Open question whether the percent should live in the label or if that is overkill.
* Our default, as with forms, should be horizontal orientation like the labels in cards example on the website. We will also need to support vertically-oriented labels.
* Dev question: should progress bars become another form field? or does it make more sense to have them be their own thing?
* Dev question: how does this play out with circular progress indicators?

**Comment by chlane** [comment]
Flash progress bars use color alone to convey success

Steps to reproduce the behavior:

1. Go to https://clarity.design/documentation/progress
2. Locate the 'Labeled With Success Flash And Fade' example
3. Activate the 'Show' button
4. Observe that the progess bar fill color changes to green to indicate success. Although not explicitly mentioned in the documention the red flash examples imply failure
. 
5. Observe that there is no non-color based indication of success or failure for the progress bars


## Expected behavior
Users who cannot see color should be able to understand the states of these progress bars. The documentation should promote providing a textual equivalent for each progress bar state.

## Versions

**App**

* Angular: NA
* Node: NA
* Clarity: NA

**Device:**

* Type: Any
* OS: Any
* Browser: Any
* Version: 2.1.2
